### DISCUSSION TITLE
Client side Theme change

### DISCUSSION THREAD

**Discussion by PMLS3** [saadeghi/daisyui#1406] [Q&A] [answered]
Hi, is it possible to update a themes colours from client-side? I.e. the client picks his own colours from front end?

**Comment by saadeghi** [comment, marked as answer]
Read this: https://github.com/saadeghi/daisyui/discussions/777

Let me know if you have any questions

**Reply by PMLS3** [reply]
> Read this: #777
> 
> Let me know if you have any questions

Thank you its working, but not for all hsl colours. Seems to be working only with HSL colours in current themes

**Reply by saadeghi** [reply]
It works for all daisyUI color names. And you can set any HSL colors (any color) to a daisyUI color name.  
Here you can see all daisyUI colors: https://daisyui.com/docs/colors/  
And here you can see how colors are getting used for themes: https://unpkg.com/browse/daisyui@2.14.3/dist/themes.css

So for example, `--p: 259 94% 51%;` is the `primary` color and `259 94% 51%` is the HSL values. You can change it to any color by changing the numbers.

Let me know if you have any questions.

**Reply by PMLS3** [reply]
Appreciate, I made an error by having commas inbetween the hsl attributes
